### ISSUE TITLE
Filename with special character "ä" triggers `stat: cannot statx` when using `occ files:scan`

### ISSUE THREAD

**Issue by chrnaum** [nextcloud/server#29418]
File "Test_ä.md" triggers "stat: cannot statx '/var/www/html/data/admin/files/Test_.md': No such file or directory" when using command "sudo docker exec --user www-data nextcloudtest_app_1 php occ files:scan admin" in ssh.

This happnes form 22.1.1 on. 22.1.0 is not affected.

### Steps to reproduce
1.
deploy stack "nextcloudtest"

```
version: '2'

services:
  dbtest:
    image: yobasystems/alpine-mariadb:latest
    command: --transaction-isolation=READ-COMMITTED --binlog-format=ROW
    restart: unless-stopped
    volumes:
      - /home/pi/nextclouddbtest:/var/lib/mysql
    environment:
      - MYSQL_ROOT_PASSWORD=rootpasswordtest
      - MYSQL_PASSWORD=passwordtest
      - MYSQL_DATABASE=nextcloudtest
      - MYSQL_USER=nextcloudtest

  app:
    image: nextcloud:22.1.1
    ports:
      - 8081:80
    links:
      - dbtest
    volumes:
      - /home/pi/nextcloudtest:/var/www/html
    restart: unless-stopped
    environment:
      - MYSQL_DATABASE=nextcloudtest
      - MYSQL_USER=nextcloudtest
      - MYSQL_PASSWORD=passwordtest
      - MYSQL_HOST=dbtest
      - NEXTCLOUD_DATA_DIR=/var/www/html/data

```
2.
In browser:
Create admin account, finish installation
Create testfile "Test_ä.md"

3.
In ssh run:
`sudo docker exec --user www-data nextcloudtest_app_1 php occ files:scan admin`

### Expected behaviour

Output 22.1.0 without error:

```
Starting scan for user 1 out of 1 (admin)
+---------+-------+--------------+
| Folders | Files | Elapsed time |
+---------+-------+--------------+
| 5       | 29    | 00:00:01     |
+---------+-------+--------------+
```

### Actual behaviour

Output 22.1.1 with error:

```
Starting scan for user 1 out of 1 (admin)
stat: cannot statx '/var/www/html/data/admin/files/Test_.md': No such file or directory
+---------+-------+--------------+
| Folders | Files | Elapsed time |
+---------+-------+--------------+
| 5       | 29    | 00:00:00     |
+---------+-------+--------------+
```

### Server configuration

**Operating system:**
Raspberry Pi 4
Raspbian GNU/Linux 10 (buster)
Kernerl: 5.10.63-v7l+

Docker version 20.10.9, build c2ea9bc

**Web server:**

**Database:**
yobasystems/alpine-mariadb:latest

**PHP version:**

**Nextcloud version:**
22.1.1

**Updated from an older Nextcloud/ownCloud or fresh install:**
fresh install

**Where did you install Nextcloud from:**
docker hub

**Signing status:**
<details>
<summary>Signing status</summary>

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
```
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

or 

Insert your config.php content here. 
Make sure to remove all sensitive content such as passwords. (e.g. database password, passwordsalt, secret, smtp password, …)
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** yes/no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

#### LDAP configuration (delete this part if not used)
<details>
<summary>LDAP config</summary>

```
With access to your command line run e.g.:
sudo -u www-data php occ ldap:show-config
from within your Nextcloud installation folder

Without access to your command line download the data/owncloud.db to your local
computer or access your SQL server remotely and run the select query:
SELECT * FROM `oc_appconfig` WHERE `appid` = 'user_ldap';


Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```
</details>

### Client configuration
**Browser:**

**Operating system:**

#### Web server error log
<details>
<summary>Web server error log</summary>

```
Insert your webserver log here
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>

```
Insert your Nextcloud log here
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>

```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```
</details>

**Comment by ei23fxg** [comment]
Yep, same here.

**Comment by acsfer** [comment]
Can't reproduce on 22.2.0

```
$ touch data/test/files/Test_ä.md

$ ./occ files:scan --all
Starting scan for user 1 out of 1 (test)
+---------+-------+--------------+
| Folders | Files | Elapsed time |
+---------+-------+--------------+
| 9       | 13    | 00:00:00     |
+---------+-------+--------------+

$ occ status
  - installed: true
  - version: 22.2.0.2
  - versionstring: 22.2.0
  - edition:
```
File is then visible on the WebUI.

**Comment by chrnaum** [comment]
Just checked with nextcloud:latest (22.2.0.2)
Here, issue still exists.


pi@raspberrypi:~ $ sudo docker exec --user www-data nextcloudtest_app_1 php occ files:scan admin
Starting scan for user 1 out of 1 (admin)
stat: cannot statx '/var/www/html/data/admin/files/Test_.md': No such file or directory
+---------+-------+--------------+
| Folders | Files | Elapsed time |
+---------+-------+--------------+
| 5       | 29    | 00:00:00     |
+---------+-------+--------------+

pi@raspberrypi:~ $ sudo docker exec --user www-data nextcloudtest_app_1 php occ status
  - installed: true
  - version: 22.2.0.2
  - versionstring: 22.2.0
  - edition:

**Comment by pzzszk** [comment]
I have the exact same issue before & after upgrading to Version 22.2.1.2.

When issuing the command "./occ files:scan --all"  everything works fine, except for the one file with a £ (pound sterling) symbol in the filename.  The folders (or groups folders) application code seems to have 'normalised' the file name  and dropped the £ symbol from the filename prior to issuing the "stat".  Therefore, it can't find the file & it issues the warning.

Somewhere this issue has already been raised - but I can't seem to find the issue number at the moment.  Previously, the issue was tracked down to something in the folders (or group folders) application code not properly recognising the locale and hence not recognising/allowing local symbols in filenames and subsequently dropping them.

My error message is:
stat: cannot stat '/mnt/md0/nextcloud/data/__groupfolders/1/Accounts/Expenditure/Expenses - SF/Foodbank receipt - 2020-12-09 - Tesco 1 x 20 Vouchers.pdf': No such file or directory

The 'real filename' for the above file is ".......Tesco 1 x £20 Vouchers.pdf"

Hope this helps with the investigation.

**Comment by tigermick** [comment]
Hi,
I have  the same issue too ...
Before, no problem with the sames files ... but recently (maybe for 22 update version) , when I try occ files:scan, I have a stat error for all file with é è à â ê ... specials characters ... for file name.
Same problem with recent 22.2.2 version

**Comment by chrnaum** [comment]
Fixed with 23.0.0.10.

pi@raspberrypi:~ $ sudo docker exec --user www-data nextcloud_app_1 php occ status
  - installed: true
  - version: 23.0.0.10
  - versionstring: 23.0.0
  - edition:
  - maintenance: false
  - needsDbUpgrade: false
  - productname: Nextcloud
  - extendedSupport: false

**Comment by fschrempf** [comment]
I'm guessing that this was fixed with #29605!? Just for reference.
